### PR TITLE
doc: rgw add some basic documentation for sync plugins & ES

### PR DESCRIPTION
**Pull request by theanalyst**
Mostly a rst formatted C-c C-v of Yehuda's mail to the ceph-devel lists

Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

### CODE REVIEW

**Comment by theanalyst** [comment]
@yehudasa ping

**Comment by theanalyst** [comment]
@yehudasa @amitkumar50 updated

**Comment by theanalyst** [comment]
jenkins test docs

**Comment by theanalyst** [comment]
jenkins test docs

File: doc/radosgw/elastic-sync-module.rst [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+          }
+        }
+      }
+
+
+
+ElasticSearch tier type configurables
+-------------------------------------
+
+* endpoint
```
Comment (theanalyst) [inline doc/radosgw/elastic-sync-module.rst]: enclose in double backticks

File: doc/radosgw/elastic-sync-module.rst [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+Multiple metadata fields must be comma seperated, a type can be forced for a
+field with a `;`. The currently allowed types are string(default), integer &
+date
+
+eg. if you want to index a custom object metadata x-amz-meta-year as int,
+x-amz-meta-date as type date and x-amz-meta-title as string, you'd do
+
+::
+   POST /mybooks?mdsearch
```
Comment (theanalyst) [inline doc/radosgw/elastic-sync-module.rst]: newline

File: doc/radosgw/elastic-sync-module.rst [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+::
+
+   GET /{bucket}?query={query-expr}
+
+request params:
+ - max-keys: max number of entries to return
+ - marker: pagination marker
+
+expression := [(]<arg> <op> <value> [)][<and|or> ...]
```
Comment (theanalyst) [inline doc/radosgw/elastic-sync-module.rst]: quote

**Comment by liewegas** [comment]
@theanalyst ping

**Comment by tchaikov** [comment]
@theanalyst ping?

**Comment by liewegas** [comment]
@theanalyst ping

**Comment by theanalyst** [comment]
@yehudasa updated

**Comment by yehudasa** [comment]
jenkins test docs

File: doc/radosgw/elastic-sync-module.rst [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+Specifies whether all user custom metadata will be indexed, or whether
+user will need to configure (at the bucket level) what custome
+metadata entries should be indexed. This is false by default
+
+* ``index_buckets_list`` (comma separated list of strings)
+
+If empty, all buckets will be indexed. Otherwise, only buckets
+specified here will be indexed. It is possible to provide bucket
+prefixes (e.g., foo*), or bucket suffixes (e.g., *bar).
```
Comment (tchaikov) [inline doc/radosgw/elastic-sync-module.rst]:
```
Warning, treated as error:
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/radosgw/elastic-sync-module.rst:66:Inline emphasis start-string without end-string.
Build step 'Execute shell' marked build as failure
```

@theanalyst  you need to escape the asterisk . like
```
prefixes (e.g., foo\*), or bucket suffixes (e.g., \*bar).
```

**Comment by theanalyst** [comment]
jenkins test docs

**Comment by theanalyst** [comment]
and docs checks are green, did some reorganization by creating another index for sync modules so that future sync modules can be added under the sync-modules tree

**Comment by theanalyst** [comment]
@yehudasa can you take a look again

**Comment by theanalyst** [comment]
jenkins render docs

**Comment by theanalyst** [comment]
@yehudasa is this a good enough first pass at the docs?

File: doc/radosgw/sync-modules.rst [hunk 5]
```diff
@@ -0,0 +1,94 @@
+============
+Sync Modules
+============
+
+.. versionadded:: Kraken
+
+The `Multisite`_ functionality of RGW introduced in Jewel allowed the ability to
+create multiple zones and mirror data & metadata between them. ``Sync Modules``
+are built atop of the multisite framework that allows for forwarding data &
```
Comment (yehudasa) [inline doc/radosgw/sync-modules.rst]: use `and` instead of `&`

Comment (theanalyst) [reply doc/radosgw/sync-modules.rst]: updated all instances

**Comment by theanalyst** [comment]
jenkins test docs

**Pull request merged**